File: ETicaretAPI/Core/ETicaretAPI.Domain/Entities/Customer.cs
using ETicaretAPI.Domain.Entities.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ETicaretAPI.Domain.Entities
{
    public class Customer: BaseEntities
    {
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public bool IsActive { get; set; }
        public DateTime? LastLoginDate { get; set; }
        public bool IsCorporate { get; set; }
        public string? CompanyName { get; set; }
        public string? TaxNumber { get; set; }
        public string? TaxOffice { get; set; }
        public int TotalOrderCount { get; set; }
        public decimal TotalSpentAmount { get; set; }
        public ICollection<Order> Orders { get; set; }
        public ICollection<Address> Addresses { get; set; }
    }
}

File: ETicaretClient/src/app/contracts/customer.ts
export class Customer {
    id: string;
    name: string;
    surname: string;
    email: string;
    phoneNumber: string;
    isActive: boolean;
    lastLoginDate?: Date;
    isCorporate: boolean;
    companyName?: string;
    taxNumber?: string;
    taxOffice?: string;
    totalOrderCount: number;
    totalSpentAmount: number;
}